comment-snyk-report/src/index.ts

```diff
@@ -143,14 +149,31 @@ ${r.paths.map((p: string) => `- \`${p}\``).join("\n")}`
 async function addOrUpdateSnykComment(commentBody: string): Promise<void> {
   const { payload, repo } = github.context;
 
-  const { data: commentsOfPR } = await octokit.rest.issues.listComments({
-    issue_number: payload.number,
-    owner: repo.owner,
-    repo: repo.repo,
-  });
+  let snykComment: { id: number } | undefined;
 
-  const snykComment = commentsOfPR.find(
-    (c: any) => c.user?.login === "github-actions[bot]"
+  await octokit.paginate(
+    octokit.rest.issues.listComments,
+    {
+      issue_number: payload.number,
+      owner: repo.owner,
+      repo: repo.repo,
+      per_page: 100,
+    },
+    (response, done) => {
+      const commentsOfPR = response.data;
+
+      snykComment = commentsOfPR.find(
+        (c: any) =>
+          c.user?.login === "github-actions[bot]" &&
+          c.body.includes("Snyk Scan Report")
```

Comment:
Let's add a top-level variable

```
const commentTitle = "🔎 Snyk Scan Report";
```

And then reuse it both here and where it's added. That would make any refactor less error-prone.

```suggestion
          c.body.includes(commentTitle)
```